model/task_lifecycle.go

```diff
@@ -1164,6 +1164,16 @@ func updateBuildStatus(b *build.Build) (bool, error) {
 
 	event.LogBuildStateChangeEvent(b.Id, buildStatus.status)
 
+	notBlockedOrUnscheduled := !buildStatus.allTasksBlocked && !buildStatus.allTasksUnscheduled
+
+	// if the status has changed, re-activate the build if it's not blocked
+	if !b.Activated && notBlockedOrUnscheduled {
```

Comment:
i don't think you have to check for activated here because SetActivated will check it again